src/js/models/layerstructure.js

```diff
@@ -1032,10 +1032,9 @@ define(function (require, exports, module) {
             }
 
             // The selected layer should be empty and a non-background layer unless replace is explicitly provided true
-            replace = replaceLayer &&
-                (replace ||
-                (!replaceLayer.isBackground && replaceLayer.isPixel &&
-                replaceLayer.bounds && !replaceLayer.bounds.area));
+            var layerShouldBeReplaced = (!replaceLayer.isBackground && replaceLayer.isPixel &&
+                    replaceLayer.bounds && !replaceLayer.bounds.area && !replaceLayer.vectorMaskEnabled);
+            replace = replaceLayer && (replace || layerShouldBeReplaced);
```

Comment:
I think the `replaceLayer` needs to be part of layerShouldBeReplaced, if `replaceLayer` is null, above code will break.